src/pages/ItemManagePage.js

```diff
@@ -0,0 +1,288 @@
+  };
+
+  // Updates the currentBatch object held within state with new
+  // values. Optional second parameter of fields to be used within
+  // the setState call.
+  updateObject = (extraObjectValues = {}, extraStateValues = {}) => {
+    const { data, currentBatch } = this.state;
+    const batchIndex = data.findIndex(({ id = 0 }) => currentBatch.id === id);
+    // If something has gone wrong finding the batch, don't try to update
+    if (batchIndex >= 0) data[batchIndex] = { ...currentBatch, ...extraObjectValues };
+    this.setState({ data: [...data], ...extraStateValues });
+  };
+
+  /**
+   * EVENT HANDLERS
+   */
+
+  // Called after entering the doses after toggling the VVM status to FAIL.
+  // Will create a new row in the table with a quantity equal to the amount
+  // entered and a VVM status of FAIL. Also updates the currentBatch objects
+  // quantity.
+  onSplitBatch = (splitValue = 0) => {
+    const { currentBatch, data } = this.state;
+    const { totalQuantity } = currentBatch;
+
+    const parsedSplitValue = parseInt(splitValue, 10);
+    let newObjectValues = {};
+
+    if (parsedSplitValue > totalQuantity) {
+      newObjectValues = { vvmStatus: false };
+      // Account for 0 & NaN (From entering a non-numeric character)
+    } else if (parsedSplitValue) {
+      const newBatchValues = { totalQuantity: parsedSplitValue, vvmStatus: false };
+      data.push(createRowObject(currentBatch, newBatchValues));
+      newObjectValues = { vvmStatus: true, totalQuantity: totalQuantity - parsedSplitValue };
+    }
+    this.updateObject(newObjectValues, { isModalOpen: false });
+  };
+
+  onApplyChanges = () => {
+    // TODO:
+    // Confirmation dialog
+    // Create inventory adjustments for any failed vvm status'
+    // Create repacks for any location changes
+    // Pop this page off the navigation stack.
+    // If more data is needed to create these repacks and inventory
+    // adjustments, they can easily be added to the row object with
+    // no side-effects. e.g. the ItemBatch itself or just
+    // some extra fields.
+  };
+
+  onDispose = () => {
+    // TODO:
+    // Open a modal to select the doses?
+    // Set the row to have some distinctive marker to show it
+    // is being disposed of.
+    // Set the dispose icon to one which will 'undelete' this row.
+  };
+
+  // Called on selecting a fridge in the fridge selection modal,
+  // just update the location of the currentBatch and close the modal.
+  onFridgeSelection = location => {
+    this.updateObject({ location }, { isModalOpen: false });
+  };
+
+  // Called on toggle the vvm toggle bar. If being set to PASS, will just update
+  // the object held in the closure to a PASS VVM status. Otherwise, opens a
+  // modal for a user to enter the doses affected and set the currentBatch.
+  onVvmToggle = currentBatch => ({ newState }) => {
+    if (!newState) return this.setState({ modalKey: 'vvmStatus', isModalOpen: true, currentBatch });
```

Comment:
you can use this.onModalUpdateHere, can even replace props with: (modalUpdateProps)

Reply:
nice 694a008cb0e11cd8545e818eaedd4a3dd3ad3c5d